src/Raven.Client/Documents/Smuggler/SmugglerResult.cs

```diff
@@ -50,7 +50,16 @@ public SmugglerResult()
 
         public IOperationProgress Progress => _progress;
 
-        public IReadOnlyList<string> Messages => _messages;
+        public IReadOnlyList<string> Messages
+        {
+            get
+            {
+                lock (this)
+                {
+                    return _messages;
```

Comment:
That is a really bad idea. You are locking, but returning the a mutable instance.